### PR TITLE
Remove uWSGI / WSGI support

### PR DESCRIPTION
**Pull request by mvdbeek**
Addresses https://github.com/galaxyproject/galaxy/issues/12257

This removes support for running Galaxy as a WSGI application (so removes both uWSGI and paster support).
To continue testing interactive tools this PR adds the possibility to run tests where Galaxy is driven by gravity (https://github.com/galaxyproject/galaxy/pull/13071/commits/945ecca943fd5a3627c0d708afbec2a1c54acce7)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by nsoranzo** [comment]
Other files that mention `scripts/paster.py`:
- [ ] `contrib/galaxy.debian-init` (update or delete?)
- [ ] `contrib/galaxy.solaris-smf.xml` (last modified in 2009, delete?)
- [ ] `doc/source/dev/starting_galaxy.md` (update?)
- [ ] `scripts/tool_shed/bootstrap_tool_shed/bootstrap_tool_shed.sh` (update?)

File: lib/galaxy/containers/docker_swarm.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def run_in_container(self, command, image=None, **kwopts):
     def _run_swarm_manager(self):
         if self._conf.managed and self._conf.manager_autostart:
             try:
-                # sys.exectuable would be preferable to using $PATH, but sys.executable is probably uwsgi
+                # sys.exectuable would be preferable to using $PATH, but sys.executable is probably gunicorn
```
Comment (nsoranzo) [inline lib/galaxy/containers/docker_swarm.py]:
```suggestion
                # sys.executable would be preferable to using $PATH, but sys.executable is probably gunicorn
```

Comment (nsoranzo) [reply lib/galaxy/containers/docker_swarm.py]: Is this comment actually still valid? We could probably replace `"python"` with `sys.executable` below.

File: lib/galaxy/util/web_compat.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 """Work around for gross circular dependency between galaxy.util and galaxy.web_stack.
```
Comment (nsoranzo) [inline lib/galaxy/util/web_compat.py]: This file can removed as it's not needed any more after commit 1497b31f0409e5de1f60dc5480a702925580860b .

File: test/unit/config/config_manage/test_config_manage.py [hunk 3]
```diff
@@ -21,10 +21,6 @@ def test_reports_conversion_1607_sample():
         reports_config = config["reports"] or {}
         assert "use_beaker_session" not in reports_config
```
Comment (nsoranzo) [inline test/unit/config/config_manage/test_config_manage.py:22]:
```suggestion
        assert "use_beaker_session" not in reports_config
        assert "uwsgi" not in config
```

File: test/unit/app/jobs/test_job_configuration.py [hunk 4]
```diff
@@ -82,19 +73,10 @@ def application_stack(self) -> ApplicationStack:
...
     def job_config(self):
         if not self._job_configuration:
-            base_handler_pools = self._job_configuration_base_pools or JobConfiguration.DEFAULT_BASE_HANDLER_POOLS
-            mock_uwsgi = mock.Mock()
-            mock_uwsgi.mule_id = lambda: 1
-            with mock.patch("galaxy.web_stack.uwsgi", mock_uwsgi), mock.patch(
-                "galaxy.web_stack.uwsgi.opt", self._uwsgi_opt
-            ), mock.patch("galaxy.jobs.JobConfiguration.DEFAULT_BASE_HANDLER_POOLS", base_handler_pools):
-                self._job_configuration = JobConfiguration(self.app)
+            self._job_configuration_base_pools or JobConfiguration.DEFAULT_BASE_HANDLER_POOLS
```
Comment (nsoranzo) [inline test/unit/app/jobs/test_job_configuration.py]: Is this line doing anything?

File: run.sh [hunk 5]
```diff
@@ -50,9 +50,6 @@ if [ "$INITIALIZE_TOOL_DEPENDENCIES" -eq 1 ]; then
     python ./scripts/manage_tool_dependencies.py init_if_needed
 fi
 
-[ -n "$GALAXY_UWSGI" ] && APP_WEBSERVER='uwsgi'
-find_server "${GALAXY_CONFIG_FILE:-none}" galaxy
```
Comment (nsoranzo) [inline run.sh:54]: I think only the first of these 2 lines should be deleted, the other is needed to set the `run_server` variable.

File: run.sh [hunk 6]
```diff
@@ -50,9 +50,6 @@ if [ "$INITIALIZE_TOOL_DEPENDENCIES" -eq 1 ]; then
...
-
 if [ "$run_server" = "python" -a -n "$GALAXY_RUN_ALL" ]; then
```
Comment (nsoranzo) [inline run.sh]: `$run_server` cannot be `python` any more, so the `if` branch here can be dropped.

File: lib/galaxy/config/config_manage.py [hunk 7]
```diff
@@ -396,7 +184,7 @@ class App(NamedTuple):
     expected_app_factories: List[str]
     destination: str
     schema_path: str
-    uwsgi_module: str
+    app_factory: str
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: It seems that this attribute is not used, drop?

File: lib/galaxy/containers/docker_swarm.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/containers/docker_swarm.py]: We discussed dropping the whole package (I think in the testing WG meeting), I don't think this is seeing any use. I'd probably do that in another PR though

File: test/unit/app/jobs/test_job_configuration.py (hunk 4 above)
Comment (mvdbeek) [reply test/unit/app/jobs/test_job_configuration.py]: Doesn't look like it, I'll drop it

**Comment by nsoranzo** [comment]
Sure, I'll give it a last check. Would you like to update the description and move the items not addressed to https://github.com/galaxyproject/galaxy/issues/12257 ?

File: lib/galaxy/config/config_manage.py [hunk 8]
```diff
@@ -674,12 +421,6 @@ def _run_conversion(args, app_desc):
                 _warn(message)
                 continue
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: I think everything from `server_section = None` till here can be removed.

File: lib/galaxy/config/config_manage.py [hunk 9]
```diff
@@ -674,12 +421,6 @@ def _run_conversion(args, app_desc):
...
 
-    if not server_section:
-        _warn("No server section found, using default uwsgi server definition.")
-        server_config = {}
-    else:
-        server_config = dict(p.items(server_section))
-
     app_items = _find_app_options_from_config_parser(p)
     applied_filters = []
     if filters:
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: We can delete also from here...

File: lib/galaxy/config/config_manage.py [hunk 10]
```diff
@@ -691,8 +432,6 @@ def _run_conversion(args, app_desc):
                     _warn(f"Unknown filter found [{value}], exiting...")
                     sys.exit(1)
```
Comment (nsoranzo) [inline lib/galaxy/config/config_manage.py]: ... to here.

File: lib/galaxy/web_stack/__init__.py [hunk 11]
```diff
@@ -220,372 +185,12 @@ def set_postfork_server_name(self, app):
...
-        facts["instance_id"] = self.instance_id
-        return facts
-
-    def shutdown(self):
-        super().shutdown()
-
-
-class PasteApplicationStack(ApplicationStack):
-    name = "Python Paste"
+class WebApplicationStack(ApplicationStack):
```
Comment (nsoranzo) [inline lib/galaxy/web_stack/__init__.py:192]: Unused?

File: lib/galaxy/web_stack/__init__.py [hunk 12]
```diff
@@ -717,25 +315,5 @@ def get_app_kwds(config_section, app_name=None):
...
-        f(*args, **kwargs)
-
-
-def _mule_fixup():
-    install_opener(None)
-
-
-if uwsgi:
-    uwsgi.post_fork_hook = _do_uwsgi_postfork
+def get_stack_facts(config=None):
```
Comment (nsoranzo) [inline lib/galaxy/web_stack/__init__.py]: Also unused?

File: lib/galaxy/web_stack/__init__.py (hunk 11 above)
Comment (mvdbeek) [reply lib/galaxy/web_stack/__init__.py:192]: Should be used by non-gunicorn asgi application servers (running my dev instance with uvicorn for instance), but I'll double check

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 13]
```diff
@@ -57,7 +57,7 @@ def app_pair(global_conf, load_app_kwds=None, wsgi_preflight=True, **kwargs):
         galaxy.app.app = app
     else:
         try:
-            app = galaxy.app.UniverseApplication(global_conf=global_conf, **kwargs)
+            app = galaxy.app.UniverseApplication(global_conf=global_conf, application_stack="fastAPI", **kwargs)
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/buildapp.py]: The new `application_stack` also doesn't seem to be used?

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**